environment/src/test/java/com/sequenceiq/environment/environment/flow/upgrade/ccm/UpgradeCcmActionsTest.java

```diff
@@ -0,0 +1,303 @@
+
+    @Captor
+    private ArgumentCaptor<String> selectorArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Event<?>> eventArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Map<String, Object>> headersArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<Object> payloadArgumentCaptor;
+
+    @Mock
+    private Tracer tracer;
+
+    @Mock
+    private Tracer.SpanBuilder spanBuilder;
+
+    @Mock
+    private Span span;
+
+    @Mock
+    private Scope scope;
+
+    @Mock
+    private SpanContext spanContext;
+
+    @Mock
+    private FlowEvent flowEvent;
+
+    @BeforeEach
+    void setUp() {
+        FlowParameters flowParameters = new FlowParameters(FLOW_ID, FLOW_TRIGGER_USER_CRN, null);
+        when(stateContext.getMessageHeader(MessageFactory.HEADERS.FLOW_PARAMETERS.name())).thenReturn(flowParameters);
+        actionPayload = new UpgradeCcmEvent(ACTION_PAYLOAD_SELECTOR, ENVIRONMENT_ID, ENVIRONMENT_NAME, ENVIRONMENT_CRN);
+        when(stateContext.getMessageHeader(MessageFactory.HEADERS.DATA.name())).thenReturn(actionPayload);
+        when(stateContext.getExtendedState()).thenReturn(extendedState);
+        when(extendedState.getVariables()).thenReturn(new HashMap<>());
+        when(stateContext.getStateMachine()).thenReturn(stateMachine);
+        when(stateMachine.getState()).thenReturn(state);
+        lenient().when(reactorEventFactory.createEvent(anyMap(), isNotNull())).thenReturn(event);
+
+        when(stateContext.getEvent()).thenReturn(flowEvent);
+        when(tracer.buildSpan(anyString())).thenReturn(spanBuilder);
+        when(spanBuilder.addReference(anyString(), any())).thenReturn(spanBuilder);
+        when(spanBuilder.ignoreActiveSpan()).thenReturn(spanBuilder);
+        when(spanBuilder.start()).thenReturn(span);
+        when(tracer.activateSpan(span)).thenReturn(scope);
+        when(span.context()).thenReturn(spanContext);
+        when(flowEvent.name()).thenReturn("eventName");
+    }
+
+    @Test
+    void validationHappyPath() {
+        testDeleteActionHappyPath(underTest::upgradeCcmValidationAction, UPGRADE_CCM_VALIDATION_HANDLER.selector(),
+                EnvironmentStatus.UPGRADE_CCM_VALIDATION_IN_PROGRESS, ResourceEvent.ENVIRONMENT_UPGRADE_CCM_VALIDATION_STARTED);
+    }
+
+    @Test
+    void validationNoEnvironment() {
+        testNoEnvironment(underTest::upgradeCcmValidationAction, UPGRADE_CCM_VALIDATION_HANDLER.selector());
+    }
+
+    @Test
+    void upgradeFreeipaHappyPath() {
+        testDeleteActionHappyPath(underTest::upgradeCcmInFreeIpaAction, UPGRADE_CCM_FREEIPA_HANDLER.selector(),
+                EnvironmentStatus.UPGRADE_CCM_ON_FREEIPA_IN_PROGRESS, ResourceEvent.ENVIRONMENT_UPGRADE_CCM_ON_FREEIPA_STARTED);
+    }
+
+    @Test
+    void upgradeFreeipaNoEnvironment() {
+        testNoEnvironment(underTest::upgradeCcmInFreeIpaAction, UPGRADE_CCM_FREEIPA_HANDLER.selector());
+    }
+
+    @Test
+    void upgradeDatalakeHappyPath() {
```

Comment:
Use camel case for "Datalake": `DataLake`.

Reply:
c'mon ;)